src/SSHDebugPS/OutputWindow.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+
+using System;
+using System.Collections.Generic;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.SSHDebugPS
+{
+    class OutputWindow
+    {
+        private static OutputWindow _instance;
+
+        public static OutputWindow Instance
+        {
+            get
+            {
+                if (_instance == null)
+                {
+                    _instance = new OutputWindow();
+                }
+
+                return _instance;
+            }
+        }
+
+        private class PaneInfo
+        {
+            public PaneInfo(Guid paneId)
+            {
+                this.paneId = paneId;
+                this.Shown = false;
+            }
+
+            internal Guid paneId;
+            internal bool Shown { get; set; }
+        }
+
+        private IVsOutputWindow outputWindow;
+        private IVsUIShell shell;
+
+        private Dictionary<string, PaneInfo> panes = new Dictionary<string, PaneInfo>()
+        {
+            // The 'Debug' pane exists by default
+            { "Debug", new PaneInfo(VSConstants.GUID_OutWindowDebugPane) }
+        };
+
+        private OutputWindow()
+        {
+            outputWindow = Package.GetGlobalService(typeof(SVsOutputWindow)) as IVsOutputWindow;
+            shell = Package.GetGlobalService(typeof(SVsUIShell)) as IVsUIShell;
+            if (outputWindow == null || shell == null)
+            {
+                throw new NullReferenceException("Could not create OutputWindow");
```

Comment:
maybe debug.assert and then no-opt the writing. I don't know if throwing is the best option here since it will crash VS.

Reply:
Also, don't throw NullReferenceException. InvalidOperationException or NotImplementedException seem to make sense.

---
In reply to: [452387943](https://github.com/microsoft/MIEngine/pull/1017#discussion_r452387943) [](ancestors = 452387943)